### PR TITLE
R68065 mattermost role pr devel 2.x

### PR DESCRIPTION
**Pull request by nfawbert**

### CODE REVIEW

File: roles/debian/mattermost/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  community.postgresql.postgresql_privs:
+    db: postgres
+    privs: ALL
+    type: database
+    obj: mattermost
+    role: mmuser
+
+- name: Reload PostgreSQL.
+  ansible.builtin.command: /usr/bin/systemctl reload postgresql
+  become: yes
```
Comment (gregharvey) [inline roles/debian/mattermost/tasks/main.yml]: The linter is right, `true` or `false`. The Ansible documentation is still littered with examples using `yes` or `no`, but that's deprecated and nobody has done a push to tidy it up.

File: roles/debian/mattermost/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  ansible.builtin.user:
+    name: mattermost
+    group: mattermost
+    system: true
+
+- name: Recursively set ownership of mattermost directory.
+  ansible.builtin.file:
+    path: /opt/mattermost
+    state: directory
+    recurse: yes
```
Comment (gregharvey) [inline roles/debian/mattermost/tasks/main.yml]: Sames, `true` please.

File: roles/debian/mattermost/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- name: Copy Mattermost systemd service def.
+  ansible.builtin.template:
+    src: "mattermost_service.j2"
+    dest: "/lib/systemd/system/mattermost.service"
+    owner: root
+    group: root
+    mode: 0644
+    force: true
+
+- name: Reload systemd units.
```
Comment (gregharvey) [inline roles/debian/mattermost/tasks/main.yml]: There's a module for doing this, you can also do it as a handler which means it won't run every time, but only if a certain task triggers it (e.g. only if we change the template file).

Comment (gregharvey) [reply roles/debian/mattermost/tasks/main.yml]: Here's an example of a handler using `ansible.builtin.service`:
* https://github.com/codeenigma/ce-provision/blob/2.x/roles/debian/openvpn/handlers/main.yml

Here's how the handler gets triggered, using `notify` - so if that task has state `changed` then it *notifies* the handler to run, and the handler will fire right at the end of all the roles:
* https://github.com/codeenigma/ce-provision/blob/2.x/roles/debian/openvpn/tasks/main.yml#L168

Here's the docs for the module:
* https://docs.ansible.com/ansible/latest/collections/ansible/builtin/service_module.html

**Review by gregharvey — CHANGES_REQUESTED** [review]

File: roles/debian/mattermost/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    path: /opt/mattermost
+    state: directory
+    recurse: true
+    owner: mattermost
+    group: mattermost
+    mode: g+w
+
+- name: Copy Mattermost systemd service def.
+  ansible.builtin.template:
+    src: "mattermost_service.j2"
```
Comment (gregharvey) [inline roles/debian/mattermost/tasks/main.yml:67]: You'll need to provide a default `templates/mattermost_service.j2` file with the role.

**Review by gregharvey — CHANGES_REQUESTED** [review]

File: roles/debian/mattermost/tasks/main.yml (hunk 1 above)
Comment (nfawbert) [reply roles/debian/mattermost/tasks/main.yml]: done

File: roles/debian/mattermost/tasks/main.yml (hunk 2 above)
Comment (nfawbert) [reply roles/debian/mattermost/tasks/main.yml]: done

File: roles/debian/mattermost/tasks/main.yml (hunk 3 above)
Comment (nfawbert) [reply roles/debian/mattermost/tasks/main.yml]: Used handlers for reloading systemd units and restarting mattermost/postfix services, here;

https://github.com/codeenigma/ce-provision/pull/1788/commits/18e6f5fd4f53f34d7d46a5d0336acf1f5c717fc8
https://github.com/codeenigma/ce-provision/pull/1788/commits/5d3c9998f2cbb5090f18af8da78918a1d434275b

File: roles/debian/mattermost/tasks/main.yml (hunk 4 above)
Comment (nfawbert) [reply roles/debian/mattermost/tasks/main.yml:67]: default systemd unit file template provided for the mattermost role here;

https://github.com/codeenigma/ce-provision/pull/1788/commits/eb6bfb0b16375878583c480f4bb86b60480a8cea

**Review by gregharvey — APPROVED** [review]

**Pull request merged**